### ISSUE TITLE
After upgrade keeps spinning

### ISSUE THREAD

**Issue by LEDfan** [owncloud/mail#1622]
### Steps to reproduce
1. Have a Nextcloud 9 instance
2. Upgrade to Nextcloud 10 RC 1
3. Open the mail app

### Expected behaviour
The accounts and mail should load.

### Actual behaviour
Spinners keeps spinning, accounts doesn't load.

### Mail app

**Mail app version:** 0.5.2

**Mailserver or service:** Small provider

**Transport security - IMAP:** SSL

**Transport security - SMTP:** SSL

**Number of accounts:** 3

### Server configuration
**Operating system**: Centos 7

**Web server:** Apache 2.4.5

**Database:** Mariadb 5.5.50

**PHP version:** 7.0.9

**Version:** Nextcloud 10

**Updated from an older version or fresh install:** Nextcloud 9

**List of activated apps:**

```
  - activity: 2.3.2
  - calendar: 1.3.2
  - comments: 1.0.0
  - contacts: 1.3.1.0
  - dav: 1.0.0
  - federatedfilesharing: 1.0.1
  - federation: 1.0.1
  - files: 1.5.2
  - files_external: 1.0.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 1.0.0
  - files_texteditor: 2.1
  - files_trashbin: 1.0.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - gallery: 15.0.0
  - mail: 0.5.2
  - news: 9.0.1
  - notes: 2.0.2
  - notifications: 0.3.0
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - richdocuments: 1.1.4
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - systemtags: 1.0.1
  - tasks: 0.9.3
  - theming: 1.0.1
  - updatenotification: 1.0.1
  - user_external: 0.4
  - workflowengine: 1.0.0
```

**The content of config/config.php:**

<details>
```
{
    "system": {
        "instanceid": "",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "
        ],
        "datadirectory": "/srv/xxx//data",
        "dbtype": "mysql",
        "version": "9.1.0.15",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "maintenance": false,
        "maxZipInputSize": 0,
        "allowZipDownload": true,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "app.mail.server-side-cache.enabled": true,
        "trashbin_retention_obligation": "auto",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": "true",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "preview_max_scale_factor": 1,
        "enabledPreviewProviders": {
            "0": "OC\\Preview\\PNG",
            "1": "OC\\Preview\\JPEG",
            "2": "OC\\Preview\\GIF",
            "11": "OC\\Preview\\Illustrator",
            "12": "OC\\Preview\\Postscript",
            "13": "OC\\Preview\\Photoshop",
            "14": "OC\\Preview\\TIFF"
        },
        "singleuser": false,
        "loglevel": 2,
        "updater.release.channel": "stable",
        "updater.server.url": "https:\/\/updates.nextcloud.org\/updater_server\/",
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "overwrite.cli.url": "https:\/\/xx.xx",
        "appstore.experimental.enabled": true
    }
}

```
</details>

#### Client configuration
**Browser:** FF 48

**Operating system:** OpenSUE

##### Server log (data/owncloud.log or data/nextcloud.log)
<details>
{"reqId":"V7g2@G9y1rMz3d@-5mwOjQAAABM","remoteAddr":"ipv6ip","app":"core","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\DriverException\",\"Message\":\"An exception occurred while executing 'SELECT `uid`, `displayname` FROM `oc_users` WHERE LOWER(`uid`) = LOWER(?)' with params [\\\"myuser\\\"]:\\n\\nSQLSTATE[HY000]: General error: 2006 MySQL server has gone away\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/myserver\\\/www\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/srv\\\/myserver\\\/www\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Statement.php(174): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'SELECT `uid`, `...', Array)\\n#2 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/db\\\/statementwrapper.php(74): Doctrine\\\\DBAL\\\\Statement->execute(Array)\\n#3 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/User\\\/Database.php(231): OC_DB_StatementWrapper->execute(Array)\\n#4 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/User\\\/Database.php(278): OC\\\\User\\\\Database->loadUser('myuser')\\n#5 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/User\\\/Manager.php(140): OC\\\\User\\\\Database->userExists('myuser')\\n#6 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/User\\\/Session.php(197): OC\\\\User\\\\Manager->get('myuser')\\n#7 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/L10N\\\/Factory.php(132): OC\\\\User\\\\Session->getUser()\\n#8 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/l10n.php(313): OC\\\\L10N\\\\Factory->findLanguage('search')\\n#9 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/util.php(511): OC_L10N::findLanguage('search')\\n#10 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/util.php(484): OC_Util::addTranslations('search')\\n#11 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/template.php(136): OC_Util::addScript('search', 'search', true)\\n#12 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/template.php(74): OC_Template::initTemplateEngine('error')\\n#13 \\\/srv\\\/myserver\\\/www\\\/lib\\\/private\\\/legacy\\\/template.php(347): OC_Template->__construct('', 'exception', 'error', false)\\n#14 \\\/srv\\\/myserver\\\/www\\\/index.php(64): OC_Template::printExceptionErrorPage(Object(Doctrine\\\\DBAL\\\\Exception\\\\DriverException))\\n#15 {main}\",\"File\":\"\\\/srv\\\/myserver\\\/www\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":116}","level":3,"time":"2016-08-20T10:55:11+00:00","method":"GET","url":"\/apps\/serverinfo\/update","user":"myuser"}
{"reqId":"V7hcrbehmlZfzkylYPEj0gAAAAc","remoteAddr":"ipv6ip","app":"core","message":"Login failed: 'myuser' (Remote IP: 'ipv6ip')","level":2,"time":"2016-08-20T13:35:41+00:00","method":"PROPFIND","url":"\/remote.php\/carddav\/addressbooks\/myuser\/contactpersonen\/","user":"--"}
{"reqId":"V7hcrbehmlZfzkylYPEj0wAAAAc","remoteAddr":"ipv6ip","app":"core","message":"Login failed: 'myuser' (Remote IP: 'ipv6ip')","level":2,"time":"2016-08-20T13:35:42+00:00","method":"PROPFIND","url":"\/remote.php\/carddav\/addressbooks\/myuser\/contactpersonen\/","user":"--"}
{"reqId":"V7hqXDVpK0CYQYwgMekFMQAAAAA","remoteAddr":"2a02:1810:280c:a300:9665:9cff:fe22:66c7","app":"core","message":"Login failed: 'myuser' (Remote IP: '2a02:1810:280c:a300:9665:9cff:fe22:66c7')","level":2,"time":"2016-08-20T14:34:04+00:00","method":"GET","url":"\/ocs\/v1.php\/apps\/files_sharing\/api\/v1\/shares","user":"--"}
{"reqId":"V7lCzc5pbZWhUfXHs5vezgAAAAM","remoteAddr":"ipv6ip","app":"files","message":"Could not find resource file \"\/apps\/files\/js\/jquery.iframe-transport.js\"","level":3,"time":"2016-08-21T05:57:34+00:00","method":"GET","url":"\/apps\/richdocuments\/index","user":"myuser"}
{"reqId":"V7lC0M5pbZWhUfXHs5ve1AAAAAM","remoteAddr":"ipv6ip","app":"PHP","message":"Undefined index: filename at \/srv\/myserver\/www\/apps\/richdocuments\/controller\/documentcontroller.php#288","level":3,"time":"2016-08-21T05:57:36+00:00","method":"POST","url":"\/apps\/richdocuments\/ajax\/documents\/create","user":"myuser"}
{"reqId":"V7lC0M5pbZWhUfXHs5ve1AAAAAM","remoteAddr":"ipv6ip","app":"PHP","message":"Undefined index: dir at \/srv\/myserver\/www\/apps\/richdocuments\/controller\/documentcontroller.php#289","level":3,"time":"2016-08-21T05:57:36+00:00","method":"POST","url":"\/apps\/richdocuments\/ajax\/documents\/create","user":"myuser"}
{"reqId":"V7lC2HSHhrpq0MoSElLhVQAAAAI","remoteAddr":"ipv6ip","app":"files","message":"Could not find resource file \"\/apps\/files\/js\/jquery.iframe-transport.js\"","level":3,"time":"2016-08-21T05:57:44+00:00","method":"GET","url":"\/apps\/richdocuments\/index","user":"myuser"}
{"reqId":"V7lC4P9ZKkCQWgLSphUsxgAAAAU","remoteAddr":"ipv6ip","app":"PHP","message":"Undefined index: filename at \/srv\/myserver\/www\/apps\/richdocuments\/controller\/documentcontroller.php#288","level":3,"time":"2016-08-21T05:57:52+00:00","method":"POST","url":"\/apps\/richdocuments\/ajax\/documents\/create","user":"myuser"}
{"reqId":"V7lC4P9ZKkCQWgLSphUsxgAAAAU","remoteAddr":"ipv6ip","app":"PHP","message":"Undefined index: dir at \/srv\/myserver\/www\/apps\/richdocuments\/controller\/documentcontroller.php#289","level":3,"time":"2016-08-21T05:57:52+00:00","method":"POST","url":"\/apps\/richdocuments\/ajax\/documents\/create","user":"myuser"}

</details>
        
##### Browser log
<details>
```
Error: An "el" #mail-new-message-fixed must exist in DOM  mail.min.js:150
```
All request have status 200
</details>

**Comment by LEDfan** [comment]
Another user having the same problem: https://github.com/owncloud/mail/issues/1622

**Comment by zeugmatis** [comment]
hrmmm this references nextcloud though, and this is the owncloud repo - not sure if that matters?

**Comment by ChristophWurst** [comment]
> Another user having the same problem: #1622

This is a reference to this very same issue ;-)

**Comment by ChristophWurst** [comment]
@LEDfan please complete the issue template and provide owncloud.log and horde-imap.log, thanks

**Comment by ChristophWurst** [comment]
> hrmmm this references nextcloud though, and this is the owncloud repo - not sure if that matters?

It doesn't matter. This app is compatible with oc and nc at the moment

**Comment by LEDfan** [comment]
@ChristophWurst thanks for looking at this :)

Here is the imap log https://gist.github.com/LEDfan/9411c243fca0b33624e6c59f63b61227
I added the owncloud.log to the opening post, but I believe you can't do much with it :P 

I just discovered something intresting, when I enable debug mode `'debug' => 'true'`, the app works. If I disable again, it doesn’t work anymore... I have tried this for 5 times.

**Comment by irgendwie** [comment]
I can now reproduce this issue with the Nextcloud 10 release:
```Error: An "el" #mail-new-message-fixed must exist in DOM```

I can also reproduce, that enabling the debug flag as a workaround solves the problem.

**Comment by ChristophWurst** [comment]
Just faced the same problem. I hope someone fixes this soon… @jancborchardt maybe?

**Comment by LukasReschke** [comment]
Yes. @jancborchardt should really fix this now.

**Comment by zeugmatis** [comment]
Working fine for me in chrome in android chrome and desktop chrome.  But Firefox 48 & 49.0b6 it is still spinning.  Clean nc install from zipfile, no upgrade: 

<img width="1230" alt="scshot" src="https://cloud.githubusercontent.com/assets/2687584/17981587/2925f7f8-6ab9-11e6-9070-a2317a819371.png">

**Comment by Draghtnod** [comment]
Same issue after upgrade from ownCloud 9.0.3 to nextCloud 10.0.0 with Firefox 48.0.2.

**Comment by LEDfan** [comment]
Please note that the error message in the console, keeps occurring when the debug mode is enabled and the app works.

**Comment by zeugmatis** [comment]
Confirmed in FF 48, enabling debug mode works, disabled debug mode does not.

**Comment by jancborchardt** [comment]
By the way @LEDfan @zeugmatis @Draghtnod since you are avid users of the Mail app you are very welcome to help testing the open pull requests! :) https://github.com/owncloud/mail/pulls

**Comment by zeugmatis** [comment]
I'll look into spinning up a dev VM next week!  Question - I'm running NC10, do I need to install OC9 dev release for this or are the codebases still similar enough, that NC10 daily build is okay?

**Comment by ChristophWurst** [comment]
fix is in https://github.com/owncloud/mail/pull/1634. Please help us test, thanks

**Comment by Gomez** [comment]
@ChristophWurst Yeah, will do the test rls today

**Comment by Gomez** [comment]
The Pre-Rls is here:
https://github.com/owncloud/mail/releases/tag/v0.5.3

Tested on oC91, please test the .tar.gz on NC10 and verify that the signature is correct. Thx

**Comment by LEDfan** [comment]
Bug is still solved with the 0.5.3 release and code integrity passes. Thanks!

**Comment by zeugmatis** [comment]
Pre-Release tested and working perfectly on NC10 as well.  Thanks!  :-) 👍
